pkg/common/peertracker/getpid.c

```diff
@@ -0,0 +1,47 @@
+#include <windows.h>
+#include <iphlpapi.h>
+#include <stdio.h>
+
+unsigned long getOwningPIDFromLocalConn(int localPort, int remotePort, int *pid)
+{
+	PMIB_TCPTABLE_OWNER_PID pTcpTable = NULL;
+	DWORD dwSize = 0;
+	unsigned long retCode = 0;
+
+	for (;;) {
+		if (dwSize > 0) {
+			pTcpTable = (PMIB_TCPTABLE_OWNER_PID)calloc(dwSize, 1);
+			if (pTcpTable == NULL) {
+				return ERROR_OUTOFMEMORY;
+			}
+		}
+
+		retCode = GetExtendedTcpTable(pTcpTable, &dwSize, TRUE, AF_INET, TCP_TABLE_OWNER_PID_ALL, 0);
+		if (retCode == NO_ERROR) {
+			break;
+		}
+
+		free(pTcpTable);

```

Comment:
Since we aren't immediately returning after this free, I think it's probably safest to set pTcpTable to NULL.
```suggestion
		free(pTcpTable);
		pTcpTable = NULL;
```